File: Data Structures & Algorithms/binary-search/submission-1.rs
impl Solution {
    pub fn search(nums: Vec<i32>, target: i32) -> i32 {
        let mut l: i32 = 0;
        let mut h: i32 = nums.len() as i32 - 1;

        while l <= h {
            let mid = l + (h-l)/2;
            if nums[mid as usize] == target {
                return mid;
            } else if nums[mid as usize] > target {
                h = mid - 1;
            } else {
                l = mid + 1;
            }
        }
        -1
    }
}

File: Data Structures & Algorithms/find-target-in-rotated-sorted-array/submission-1.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        left = 0
        right = len(nums) - 1
        pivot = 0
        while left < right:
            m = (left + right) // 2
            if nums[m] > nums[right]:
                left = m + 1
            else:
                right = m

        pivot = left

        l, r = 0, len(nums) - 1

        if target >= nums[pivot] and target <= nums[r]:
            l = pivot
        else:
            r = pivot - 1
        
        while l <= r:
            m = (l+r)//2
            if nums[m] == target:
                return m
            elif nums[m] < target:
                l = m + 1
            else:
                r = m - 1
            
             
                 
             
        return -1
        